clippy_lints/src/doc.rs

```diff
@@ -485,6 +485,8 @@ fn check_doc<'a, Events: Iterator<Item = (pulldown_cmark::Event<'a>, Range<usize
     let mut in_heading = false;
     let mut is_rust = false;
     let mut edition = None;
+    let mut ticks_unbalanced = false;
```

Comment:
You currently check for unbalanced backticks in the complete doc comment. I think it would make sense to only lint single paragraphs at once. You can use the `Start(_)` and `End(_)` events for that.

Example, this should only span the lint over B and check A as usual:
```rs
/// A
/// A
///
/// B` <- backtick
/// B
```

The bottom block that calls `check_text(...)` would also need to be moved.